scipy/integrate/tests/test_quadrature.py

```diff
@@ -397,3 +397,240 @@ def func(x):
+                    ]
+                ),
+            ),
+            (
+                # Multi dimensional values for initial and dx with non-default axis,
+                # and mixture of int and float inputs
+                np.array([[1, 4, 9, 16], [1, 4, 9, 16], [1, 4, 9, 16]]),
+                [[1.0, 2.0, 1.0, 5.0]],
+                [[0, 0, 0, 1]],
+                0,
+                np.array(
+                    [
+                        [0, 0, 0, 1],
+                        [1, 8, 9, 81],
+                        [2, 16, 18, 161],
+                    ]
+                ),
+            ),
+            (
+                # Test case with cubic y
+                np.array([[0, 1, 8]]),
+                1,
+                None,
+                None,
+                np.array([[0.0, 4.0]]),
+            ),
+        ],
+    )
+    def test_simpson_multi_dx(self, y, dx, initial, axis, int_y):
+        kwargs = {}
+        if dx is not None:
+            kwargs["dx"] = dx
+        if initial is not None:
+            kwargs["initial"] = initial
+        if axis is not None:
+            kwargs["axis"] = axis
+        assert_almost_equal(
+            cumulative_simpson(y, **kwargs),
+            int_y,
+        )
+
+    @pytest.mark.parametrize(
+        ("y_func", "x", "dx", "initial", "axis", "int_y_func"),
+        [
+            (
+                # Non-unique values in x
+                lambda y: y**2,
+                np.array([2, 2, 4, 8]),
+                None,
+                8 / 3,
+                -1,
+                lambda y: y**3 / 3,
+            ),
+            (
+                # Multi dimensional, non-unique values in x
+                lambda y: y**2,
+                np.array([[1, 2, 3, 4], [0, 2, 4, 8], [2, 2, 4, 8]]),
+                None,
+                np.array([[1 / 3], [0], [8 / 3]]),
+                -1,
+                lambda y: y**3 / 3,
+            ),
+            (
+                # Multi dimensional, non-unique values in x in non-default axis
+                lambda y: y**2,
+                np.array([[1, 2, 3, 4], [1, 2, 4, 8], [1, 2, 4, 8]]),
+                None,
+                None,
+                0,
+                lambda y: y**3 / 3,
+            ),
+            (
+                # Less than 3 points along axis of integration
+                lambda y: y**2,
+                np.array([[1, 2], [1, 2], [1, 2]]),
+                None,
+                None,
+                -1,
+                lambda y: y**3 / 3,
+            ),
+        ],
+    )
+    def test_simpson_exceptions(self, y_func, x, dx, initial, axis, int_y_func):
+        input_x = x
+        if x is None:
+            x = dx * np.arange(17)
+        y = y_func(x)
+        int_y = int_y_func(x)
+        if initial is None:
+            y_shape = [slice(None)] * y.ndim
+            y_shape[axis] = slice(1, None)
+            int_y = int_y[tuple(y_shape)]
+
+        with pytest.raises(ValueError):
```

Comment:
Use a match statement to be sure the correct exception is raised.

Reply:
Addressed in [be6e1d7](https://github.com/scipy/scipy/pull/18151/commits/be6e1d75e7863d46c297ad24a9929392873dcf3a)

Reply:
I am a bit confused, the match don't correspond to the error messages.

Reply:
Did not fully understand you. Could you explain the issue with an example?

I added a few more unit tests to cover all the ValueErrors that are raised by the function.

Reply:
What I mean is that, your function has raise statements. In the tests, we should be able to see all of these statements being verified (we only want to catch the new ones, not things coming from elsewhere.) Unless I am mistaking, I could not find some of the new exception messages in the tests.

Reply:
Got it, you were correct, not all the raise statements were covered. 
I have added more tests to address all the statements in [e5de535](https://github.com/scipy/scipy/pull/18151/commits/e5de5352353ba28a62044cd6708b00ca26b7c002)

Reply:
Thank you, great 👍 Now I would only add tests for the new exceptions. So if you have 5 new exceptions, there should just be 5 tests.

Reply:
I have added 8 exception tests for 6 new exceptions. 2 additional unit tests are required to ensure one of the raises is working for different types of input parameters (1D vs multi-D inputs and non-default integration axis).